src/connectors/snowflake/trulens/connectors/snowflake/utils/sis_dashboard_artifacts.py

```diff
@@ -0,0 +1,125 @@
+import glob
+import os
+import tempfile
+from typing import Optional, Any
+
+from trulens.connectors.snowflake.utils.server_side_evaluation_artifacts import (
+    _STAGE_NAME as _PKG_STAGE_NAME,
+)
+
+from snowflake.snowpark import Session
+
+_STAGE_NAME = "TRULENS_DASHBOARD_STAGE"
+_STREAMLIT_NAME = "TRULENS_STREAMLIT_DASHBOARD"
+_STREAMLIT_ENTRYPOINT = "Leaderboard.py"
+
+_TRULENS_DEPENDENCIES = [
+    "trulens-core",
+    "trulens-dashboard",
+    "trulens-connectors-snowflake",
+]
+
+
+class SiSDashboardArtifacts:
+    """This class is used to set up Snowflake artifacts for launching the dashboard on SiS."""
+
+    def __init__(
+        self,
+        session: Session,
+        database: str,
+        schema: str,
+        warehouse: str,
+        use_staged_packages: bool,
+    ) -> None:
+        self._session = session
+        self._database = database
+        self._schema = schema
+        self._warehouse = warehouse
+        self._use_staged_packages = use_staged_packages
+
+    def set_up_all(self) -> None:
```

Comment:
why have this `set_up_all` passthrough instead of calling `_set_up_stage` and `_set_up_streamlit` individually? This seems unneccessarily obscure

Reply:
It gives a single public entrypoint that defines how to setup the dashboard artifacts. If something changes (either with Trulens or Snowflake), the logic can be modified in one spot instead of repeating ourselves anywhere we may want to invoke this setup.